src/event_server.cpp

```diff
@@ -2087,6 +2087,47 @@ static void get_calibration_data(JObj& response, const json_value *params)
     response << jrpc_result(rslt);
 }
 
+static void set_cooler_state(JObj& response, const json_value *params)
+{
+    Params p("enabled", params);
+    const json_value *val = p.param("enabled");
+    bool enable;
+    if (!val || !bool_param(val, &enable))
+    {
+        response << jrpc_error(JSONRPC_INVALID_PARAMS, "expected enabled boolean param");
+        return;
+    }
+
+    if (!pCamera || !pCamera->Connected)
+    {
+        response << jrpc_error(1, "camera not connected");
+        return;
+    }
+
+    if (!pCamera->HasCooler)
+    {
+        response << jrpc_error(1, "camera lacks a cooler");
+        return;
+    }
+
+    if (!pCamera->SetCoolerOn(enable))
+    {
+        response << jrpc_error(1, "failed to set cooler state");

```

Comment:
```suggestion
        response << jrpc_error(1, "failed to set cooler state");
        return;
```

Reply:
function is `void` and none of the existing `void`-typed method handler funcs explicitly `return` at the end. Should this be expanded to cover the preexisting functions as well, or just this one?

Reply:
sorry, I should have been more clear on why I made the suggestion. If the SetCoolerOn call fails and we generate an error response (line 2115), then this method is done and needs to return at this point and not keep going -- if it kept going it would append additional failure or success info to the response which would be wrong.

Reply:
Oh! Right. I couldn't quite tell the line in this late hour. Fixed.